If legacy MAP, use default poster image instead of image from page data

karinathomasbbc opened the pull request:
Resolves #5591

**Overall change:**
- Check if the asset is a legacy MAP, and if so, set the image locator to null. Downstream, when this is used to populate the `og:image` metadata tag, a null image locator means that the default image will be used e.g. https://news.files.bbci.co.uk/ws/img/logos/og/arabic.png

The way in which we have configured ichef + image branding to work with CPS images, would require a large set of code changes to use the image index from the page data, which is not really feasible for TC2 MAP pages, given that traffic is low & content is rarely shared.

**Code changes:**
- if legacy MAP, return null as image locator
- Updating snapshots for integration tests

---

- [ ] I have assigned myself to this PR and the corresponding issues
- [ ] I have added the `cross-team` label to this PR if it requires visibility across World Service teams
- [ ] I have assigned this PR to the Simorgh project

**Testing:**

- [ ] Automated (jest and/or cypress) tests added (for new features) or updated (for existing features)
- [ ] If necessary, I have run the local E2E non-smoke tests relevant to my changes (`CYPRESS_APP_ENV=local CYPRESS_SMOKE=false npm run test:e2e:interactive`)
- [ ] This PR requires manual testing

karinathomasbbc committed:
If legacy MAP, use default poster image instead of image from page data, by setting the image locator to null.

The way in which we have configured ichef + image branding to work with CPS images, and would require a large change, which is not really feasible for TC2 MAP pages, as traffic is low & content is rarely shared.

karinathomasbbc committed:
Merge branch 'latest' into tc2-poster-image

RichardPK commented on src/app/pages/MediaAssetPage/MediaAssetPage.jsx:
```diff
@@ -53,9 +53,18 @@ const MediaAssetPage = ({ pageData }) => {
+  const getIndexImageLocator = () => {
+    const imageParts = indexImagePath && getImageParts(indexImagePath);
+
+    if (isLegacyMediaAssetPage(requestContext.canonicalLink)) {
```
Could bump this to line 58 so it gets hit first in the function & only does the getImageParts logic if it isn't a legacy asset? Marginal improvement 🤷‍♂️

karinathomasbbc replied:
Good shout, will do that

karinathomasbbc committed:
Merge branch 'latest' into tc2-poster-image

karinathomasbbc committed:
Merge branch 'latest' into tc2-poster-image

karinathomasbbc committed:
Merge branch 'latest' into tc2-poster-image

karinathomasbbc committed:
Reshuffle function - only get the image locator if not a legacy MAP

karinathomasbbc committed:
Merge branch 'tc2-poster-image' of github.com:bbc/simorgh into tc2-poster-image

rebeccamcginn commented on src/app/pages/MediaAssetPage/MediaAssetPage.jsx:
```diff
@@ -52,10 +52,18 @@ const MediaAssetPage = ({ pageData }) => {
+    }
+
+    const indexImagePath = path(['promo', 'indexImage', 'path'], pageData);
+    const imageParts = indexImagePath && getImageParts(indexImagePath);
```
This is quite hard to read. If we check that indexImagePath exists before line 62, we can get rid of the checks on lines 62 & 63

karinathomasbbc replied:
Could you suggest some pseudocode which would make it easier to read?

rebeccamcginn replied:
Check if indexImagePath has a value. If it doesn't, return early. Then remove the `indexImagePath &&` and `imageParts &&` checks as they will no longer be needed

karinathomasbbc committed:
Merge branch 'latest' into tc2-poster-image

karinathomasbbc committed:
Make logic more readable

karinathomasbbc committed:
Merge branch 'tc2-poster-image' of github.com:bbc/simorgh into tc2-poster-image

ryanmccombe commented on src/app/pages/MediaAssetPage/MediaAssetPage.jsx:
```diff
@@ -52,10 +52,26 @@ const MediaAssetPage = ({ pageData }) => {
-    ? getImageParts(indexImagePath)[1]
-    : null;
+
+  const getIndexImageLocator = () => {
+    if (isLegacyMediaAssetPage(requestContext.canonicalLink)) {
+      return null;
+    }
+
+    const indexImagePath = path(['promo', 'indexImage', 'path'], pageData);
+
+    if (indexImagePath) {
+      const imageParts = getImageParts(indexImagePath);
+
+      if (imageParts && imageParts.length >= 2) {
+        return imageParts[1];
+      }
+    }
+
+    return null;
+  };
+
+  const indexImageLocator = getIndexImageLocator();
```
Ramda could simplify this a lot, something like:

```javascript
const getIndexImageLocator = pipe(
  pathOr('', ['promo', 'indexImage', 'path']),
  getImageParts,
  last,
);

const indexImageLocator = isLegacyMediaAssetPage ? null : getIndexImageLocator(pageData);
```

karinathomasbbc replied:
I'll give it a shot to see what happens 😬

karinathomasbbc replied:
`pipe` didn't quite work - but I used `last` instead. I think it's _slightly_ easier to read now?

karinathomasbbc committed:
Simplify(?) logic by using ramda functions
- pathOr to avoid having to do a null check on indexImagePath
- last to get last element of array (safe for empty / null array)

karinathomasbbc committed:
Remove import as it is no longer required

ryanmccombe reviewed (approved):
👍

ryanmccombe committed:
Merge branch 'latest' into tc2-poster-image

karinathomasbbc committed:
Merge branch 'latest' into tc2-poster-image

karinathomasbbc committed:
Merge branch 'latest' into tc2-poster-image

karinathomasbbc committed:
Merge branch 'latest' into tc2-poster-image
